### DISCUSSION TITLE
Difficulties with multi interface nodes

### DISCUSSION THREAD

**Discussion by lmadarassy** [k3s-io/k3s#7280] [Q&A]
**Environmental Info:**
K3s Version:

k3s version v1.24.9+k3s2 (0d4e64f7)

Node(s) CPU architecture, OS, and Version: 

Linux myhost 5.14.21-150400.24.55-default #1 SMP PREEMPT_DYNAMIC Mon Mar 27 15:25:48 UTC 2023 (cc75cf8) x86_64 x86_64 x86_64 GNU/Linux
Cluster Configuration:

3 master nodes, disabled builtin servicelb, traefik and flannel and using calico, metallb, nginx-ingress

**Describe the bug:**
All the nodes have 3 interfaces:
eth0: using for kubernetes api access
bond0: want to use for calico interface for the CNI traffic between the nodes (using calico with vxlan)
eth4: using with metallb for bgp peering

I have installed k3s without interface specification and by default it is using eth0 and calico also using this interface and I can't find any options to change it to different interface (found only for BGP mode).
I opened this ticket here, because I think the problem is in my k3s installation options. Do you have any idea how can I do this?

Thanks,
Laszlo

**Comment by brandond** [comment]
You're probably going to have to tackle each of these asks individually with calico and metallb. You can put K3s on the preferred interface for the apiserver with the --node-ip flag, but the rest is up to you to manage - especially given that K3s doesn't include calico or metallb so you 100% own the configuration of those pieces.

**Comment by lmadarassy** [comment]
Thanks! Trying the --node-ip flag. I have tried the --kube-apiserver-arg=--advertise-address= flag, the node ip remained the eth0 ip, de it was enough for calico to use that interface.

**Comment by lmadarassy** [comment]
--node-ip seems to work, but there is 1 side effect: the kubeconfig client certificate is only valid for the node ip (+ 127.0.0.1, cluster ip, etc) and not valid for my eth0 ip, what I want to use for kuberenetes api access.
Are there any solution for this?

**Reply by brandond** [reply]
--tls-san
